packages/core/src/routes/interaction/verifications/identifier-verification.ts

```diff
@@ -1,65 +1,139 @@
+import type { Profile, SocialConnectorPayload } from '@logto/schemas';
+import { Event } from '@logto/schemas';
 import type { Provider } from 'oidc-provider';
 
-import type { InteractionContext, Identifier } from '../types/index.js';
+import RequestError from '#src/errors/RequestError/index.js';
+import { findSocialRelatedUser } from '#src/lib/social.js';
+import { verifyUserPassword } from '#src/lib/user.js';
+import assertThat from '#src/utils/assert-that.js';
+import { maskUserInfo } from '#src/utils/format.js';
+
+import type { PasswordIdentifierPayload, PasscodeIdentifierPayload } from '../types/guard.js';
+import type { InteractionContext, Identifier, SocialIdentifier } from '../types/index.js';
+import findUserByIdentifier from '../utils/find-user-by-identifier.js';
 import { isPasscodeIdentifier, isPasswordIdentifier, isProfileIdentifier } from '../utils/index.js';
+import { assignIdentifierVerificationResult } from '../utils/interaction.js';
 import { verifyIdentifierByPasscode } from '../utils/passcode-validation.js';
 import { verifySocialIdentity } from '../utils/social-verification.js';
-import {
-  verifyUserByIdentityAndPassword,
-  verifyUserByVerifiedPasscodeIdentity,
-  verifyUserBySocialIdentity,
-} from '../utils/verify-user.js';
 
-export default async function identifierVerification(
+const passwordIdentifierVerification = async (
+  identifier: PasswordIdentifierPayload
+): Promise<Identifier[]> => {
+  // TODO: Log
+  const { password, ...identity } = identifier;
+  const user = await findUserByIdentifier(identity);
+  const verifiedUser = await verifyUserPassword(user, password);
+
+  const { isSuspended, id } = verifiedUser;
+  assertThat(!isSuspended, new RequestError({ code: 'user.suspended', status: 401 }));
+
+  return [{ key: 'accountId', value: id }];
+};
+
+const passcodeIdentifierVerification = async (
+  payload: { event: Event; identifier: PasscodeIdentifierPayload; profile?: Profile },
   ctx: InteractionContext,
   provider: Provider
-): Promise<Identifier[]> {
-  const { identifier, event, profile } = ctx.interactionPayload;
+): Promise<Identifier[]> => {
+  const { identifier, event, profile } = payload;
+  const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
 
-  if (!identifier) {
-    return [];
-  }
+  await verifyIdentifierByPasscode({ ...identifier, event }, jti, ctx.log);
 
-  if (isPasswordIdentifier(identifier)) {
-    const accountId = await verifyUserByIdentityAndPassword(identifier);
+  const verifiedPasscodeIdentifier: Identifier =
+    'email' in identifier
+      ? { key: 'verifiedEmail', value: identifier.email }
+      : { key: 'verifiedPhone', value: identifier.phone };
 
-    return [{ key: 'accountId', value: accountId }];
+  // Return the verified identity directly if it is new profile identities
+  if (isProfileIdentifier(identifier, profile)) {
+    return [verifiedPasscodeIdentifier];
   }
```

Comment:
curious why we do not care about user.id in the case that identifier is a new profile identity.
Another question is -- are all the profile identities verified?